Assets/Scripts/Ground.cs

```diff
@@ -89,17 +95,18 @@ private void OnCollisionEnter(Collision other)
             PaintSurface(false);
         }
         var dir = ReturnDirection(other.gameObject, this.gameObject);
-        if ((_originalColour == Color.cyan) && other.gameObject.CompareTag("Player"))
+        if (_isIceBlockEffectEnabled && (_originalColour == Color.cyan) && other.gameObject.CompareTag("Player"))
```

Comment:
Combine this if with the one below by storing the conditions in variables (or maybe just one bool variable). It should improve readability